test/e2e/framework/addon/vault/setup.go

```diff
@@ -80,10 +80,10 @@ func NewVaultServiceAccount(name string) *corev1.ServiceAccount {
 	}
 }
 
-func NewVaultServiceAccountRole(namespace string) *rbacv1.ClusterRole {
+func NewVaultServiceAccountRole(namespace, serviceAccountName string) *rbacv1.ClusterRole {
 	return &rbacv1.ClusterRole{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: fmt.Sprintf("auth-delegator:%s:vault", namespace),
+			Name: fmt.Sprintf("auth-delegator:%s:%s", namespace, serviceAccountName),
```

Comment:
question: what is happening here?

Reply:
We are allowing custom service accounts to be added as the role to vault.

This means we can test against ClusterIssuers, which would otherwise have a conflict on the ServiceAccount name "vault" in the cluster resource namespace.

Reply:
Thank you!!